internal/flags/apiflags/flags.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apiflags
+
+import (
+	"fmt"
+	"strings"
+
+	gencrd "github.com/operator-framework/operator-sdk/internal/generate/crd"
+	"github.com/operator-framework/operator-sdk/internal/util/projutil"
+	"github.com/spf13/pflag"
+)
+
+// AddAPIFlags ...
+type AddAPIFlags struct { //nolint:golint
+
+	APIVersion       string
+	Kind             string
+	OperatorType     string
+	SkipGeneration   bool
+	Repo             string
+	GeneratePlaybook bool
+	HeaderFile       string
+	GitInit          bool
+	MakeVendor       bool
+	SkipValidation   bool
+	CrdVersion       string
+
+	HelmChartRef     string
+	HelmChartVersion string
+	HelmChartRepo    string
```

Comment:
IHMO the Helm specific flags should be in the other structure such as `HelmApiFlags`.

Reply:
See for example: https://github.com/operator-framework/operator-sdk/blob/master/pkg/helm/flags/flag.go#L34

Reply:
Improvement: This falls under improvement too, as for `operator-sdk new` , we seems not to be differentiating flags, and same has been applied to `operator-sdk add api` as well.
@joelanford WDYT

Reply:
I agree that we should make a `HelmAPIFlags` struct. I would say let's create an issue for this and have someone fix it in another PR. Proposed title for the github issue can be "Move Helm specific flags out of APIFlags into their own struct: HelmAPIFlags"

Reply:
IMO, we should hold off on further improvements to this code since it will be re-worked after the Kubebuilder integration is complete.